instrumentation/scala-2.13.0/src/main/java/scala/concurrent/impl/Transformation_Instrumentation.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package scala.concurrent.impl;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.agent.bridge.Transaction;
+import com.newrelic.api.agent.Segment;
+import com.newrelic.api.agent.Token;
+import com.newrelic.api.agent.Trace;
+import com.newrelic.api.agent.weaver.NewField;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import com.nr.agent.instrumentation.scala.ScalaUtils;
+import scala.Function1;
+import scala.concurrent.ExecutionContext;
+import scala.util.Try;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.nr.agent.instrumentation.scala.ScalaUtils.scalaFuturesAsSegments;
+
+@Weave(originalName = "scala.concurrent.impl.Promise$Transformation")
+public class Transformation_Instrumentation<T>  {
+    private Try<T> _arg = Weaver.callOriginal();
+    @NewField
+    private Token token;
+    @NewField
+    private AgentBridge.TokenAndRefCount tokenAndRefCount;
+
+    private Transformation_Instrumentation(final Function1 _fun, final ExecutionContext _ec, final Try<T> _arg, final int _xform)  {
```

Comment:
Hi @junder31, I think you could remove this weave method. We instrumented the constructor in 2.12 due to an issue with Akka-http linking. It would also allow removing the token newfield and @Trace on submitWithValue, the latter of which is causing the ProducerConsumerWithFutures.testProducerConsumer test to fail with extra segments.

Reply:
@tspring I made the changes you suggested.  I'll give it a shot on one of my apps and see if instrumentation is still working.  The test case at least passing now.

Reply:
@tspring looks like I'm still getting the transactions segments I was expecting on my app as well with these changes.